[yalantinglibs] Build error on x64-windows-static

Package: yalantinglibs:x64-windows-static@0.4.0

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.43.34808.0
-    vcpkg-tool version: 2025-02-11-bec4296bf5289dc9ce83b4f5095943e44162f9c2
    vcpkg-scripts version: unknown

**To Reproduce**

`vcpkg install yalantinglibs`

**Failure logs**

```
Downloading https://github.com/alibaba/yalantinglibs/archive/0.4.0.tar.gz -> alibaba-yalantinglibs-0.4.0.tar.gz
error: https://github.com/alibaba/yalantinglibs/archive/0.4.0.tar.gz: failed: status code 300
note: If you are using a proxy, please ensure your proxy settings are correct.
Possible causes are:
1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable to `https//address:port`.
This is not correct, because `https://` prefix claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr, etc...) is an HTTP proxy.
Try setting `http://address:port` to both HTTP_PROXY and HTTPS_PROXY instead.
2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings set by your proxy software. See: https://github.com/microsoft/vcpkg-tool/pull/77
The value set by your proxy might be wrong, or have same `https://` prefix issue.
3. Your proxy's remote server is our of service.
If you believe this is not a temporary download server failure and vcpkg needs to be changed to download this file from a different location, please submit an issue to https://github.com/Microsoft/vcpkg/issues
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:136 (message):
  Download failed, halting portfile.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_from_github.cmake:106 (vcpkg_download_distfile)
  ports/yalantinglibs/portfile.cmake:3 (vcpkg_from_github)
  scripts/ports.cmake:206 (include)

```
Installing 3/3 yalantinglibs:arm64-osx@0.4.0...
Building yalantinglibs:arm64-osx@0.4.0...
Downloading https://github.com/alibaba/yalantinglibs/archive/0.4.0.tar.gz -> alibaba-yalantinglibs-0.4.0.tar.gz
/Users/jackyu/Documents/vcpkg/downloads/alibaba-yalantinglibs-0.4.0.tar.gz.4990.part: error: download from https://github.com/alibaba/yalantinglibs/archive/0.4.0.tar.gz had an unexpected hash
note: Expected: df45fb79713b3999ccac67228c886017bd7b4088eeb48aa2334f4861caf38d4595076149ce3c5a40858cb6cb9f1ed2588c522cc70e591e66ac19c23ab812e75b
note: Actual  : 9aadd4d93b542e276fbbba284fc792fa2821a2241cb4924b2aa0be75d9f411a89602ef818cacde6e1e86600a63f74c11a42cdc7c04cf443b0854f10da74f41bc
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:136 (message):
  Download failed, halting portfile.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_from_github.cmake:106 (vcpkg_download_distfile)
  ports/yalantinglibs/portfile.cmake:3 (vcpkg_from_github)
  scripts/ports.cmake:206 (include)


error: building yalantinglibs:arm64-osx failed with: BUILD_FAILED
See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
Elapsed time to handle yalantinglibs:arm64-osx: 1.7 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+yalantinglibs
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[yalantinglibs]+Build+error+on+arm64-osx&body=Copy%20issue%20body%20from%20%2FUsers%2Fjackyu%2FDocuments%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md
```
On my MacBook.

--- a/ports/yalantinglibs/portfile.cmake
+++ b/ports/yalantinglibs/portfile.cmake
@@ -3,8 +3,8 @@ set(VCPKG_BUILD_TYPE release)  # header-only
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO alibaba/yalantinglibs
-    REF "${VERSION}"
-    SHA512 df45fb79713b3999ccac67228c886017bd7b4088eeb48aa2334f4861caf38d4595076149ce3c5a40858cb6cb9f1ed2588c522cc70e591e66ac19c23ab812e75b
+    REF 75b2c25eb591e0ee6dd94e3c764f99af761fb6f1 # Use commit id to avoid target to multiple commits
+    SHA512 8a5e01414389084a856c1ce07ae07991ecfe24b61fef9629aff531d303fd9dbb0da4fca4cd259a757788815cd8f3039fd115d5a2548845438665f6baed89163c
     HEAD_REF main
 )
 

--- a/ports/yalantinglibs/vcpkg.json
+++ b/ports/yalantinglibs/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "yalantinglibs",
   "version": "0.4.0",
+  "port-version": 1,
   "description": "A Collection of C++20 libraries, include struct_pack, struct_json, struct_xml, struct_yaml, struct_pb, easylog, coro_rpc, coro_http and async_simple",
   "homepage": "https://github.com/alibaba/yalantinglibs",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -10254,7 +10254,7 @@
     },
     "yalantinglibs": {
       "baseline": "0.4.0",
-      "port-version": 0
+      "port-version": 1
     },
     "yaml-cpp": {
       "baseline": "0.8.0",

--- a/versions/y-/yalantinglibs.json
+++ b/versions/y-/yalantinglibs.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "f82ff90e6d0979130f2e60e348f7a61a1c318c66",
+      "version": "0.4.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "18c61b804c7fba225b1db50253a9f003a34f1199",
       "version": "0.4.0",
       "port-version": 0

